app/Fission/Internal/Orphanage/RIO.hs

```diff
@@ -12,19 +12,137 @@ import qualified RIO.ByteString.Lazy as Lazy
 import           Servant.Client
 import           Network.HTTP.Client as HTTP
 
+import           Network.AWS
+import qualified Network.AWS.Auth as AWS
+import           Network.AWS.Route53
+
 import           Network.IPFS
 import           Network.IPFS.Types         as IPFS
 import qualified Network.IPFS.Process.Error as Process
 import           Network.IPFS.Process
 import qualified Network.IPFS.Peer as Peer
 
+import           Fission.AWS.Route53.Class
+import           Fission.AWS.Types as AWS
+import           Fission.AWS
 import qualified Fission.Config as Config
+import           Fission.Internal.UTF8
+import           Fission.IPFS.DNSLink
+
+instance (Has AWS.AccessKey cfg, Has AWS.SecretKey cfg) => MonadAWS (RIO cfg) where
+  liftAWS awsAction = do
+    accessKey :: AWS.AccessKey <- Config.get
+    secretKey :: AWS.SecretKey <- Config.get
+
+    env <- newEnv <| FromKeys accessKey secretKey
+
+    awsAction
+      |> runAWS env
+      |> runResourceT
+
+instance
+  ( Has AWS.AccessKey          cfg
+  , Has AWS.SecretKey          cfg
+  , Has AWS.ZoneID             cfg
+  , Has AWS.Route53MockEnabled cfg
+  , HasLogFunc                 cfg
+  )
+  => MonadRoute53 (RIO cfg) where
+  update recordType (AWS.DomainName domain) content = do
+    AWS.Route53MockEnabled mockRoute53 <- Config.get
+
+    if mockRoute53
+      then changeRecordMock
+      else changeRecord'
+
+    where
+      changeRecordMock = do
+          mockTime <- currentTime
+
+          let
+            mockMessage = mconcat
+              [ "MOCK: Updating DNS "
+              , show recordType
+              , " record at: "
+              , show domain
+              , " with "
+              , show content
+              ]
+
+            mockId             = "test123"
+            mockChangeInfo     = changeInfo mockId Pending mockTime
+            mockRecordResponse = changeResourceRecordSetsResponse 300 mockChangeInfo
+
+          logDebug mockMessage
+          return (Right mockRecordResponse)
+
+      changeRecord' = do
+        logDebug <| "Updating DNS record at: " <> displayShow domain
+
+        req <- createChangeRequest
+
+        within NorthVirginia do
+          res <- send req
+          return <| validate res
```

Comment:
💅 Having this inside of a where as opposed to their own functions feels like were misusing `where` and they would better fit as their own functions. But who knows maybe thats just the haskell way :)

Reply:
Good instinct! So, it depends. We can break them out, but it probably won't have much benefit. Here's why:

# The Big Reason

The big reason why this is special: it's typeclass instance (Elixir's `defimpl`). We don't expect to be able to reuse the functions separately at all — the point is abstraction. We can break them out, but it won't really help us.

# General Discussion

It common to use `where` to contain larger functions, but that's not a blanket statement. I agree _in general_ that keeping them small is a good idea.

`where`s are inner functions, so they have access to the arguments of the named function. This means less writing out parameters to link things together, and more compiler specialization (not that this is something that we're optimizing for).

In this case we really don't care about the mock function; it's there for local development only. Since we're not exporting it from the module, they're roughly equivalent except for the below.

Reply:
Also a great explantation. Thanks Brooke.

I think my "style" preference is still a separate function. But there's no reason other than I like saving on indentation :P